lib/core/src/Cardano/Wallet.hs

```diff
@@ -646,11 +655,13 @@ newWalletLayer tracer block0 feePolicy db nw tl = do
         -> Passphrase "encryption"
         -> CoinSelection
         -> ExceptT ErrSignTx IO (Tx t, TxMeta, [TxWitness])
-    _signTx wid pwd (CoinSelection ins outs chgs) = DB.withLock db $ do
+    _signTx wid pwd sel@(CoinSelection ins outs chgs) = DB.withLock db $ do
         (w, _) <- withExceptT ErrSignTxNoSuchWallet $ _readWallet wid
         let (changeOuts, s') = flip runState (getState w) $ forM chgs $ \c -> do
                 addr <- state genChange
                 return $ TxOut addr c
+        withExceptT (ErrSignTx . ErrInvalidTx) $
+            except $ (validateSelection tl) sel
```

Comment:
I think this is now in the wrong place (should be part of `mkStdTx` function), will confirm while reviewing the rest.